### ISSUE TITLE
Continued Pretraining on Llama 7b.

### ISSUE THREAD

**Issue by wiseyy** [huggingface/nanotron#79]
In continuation to https://github.com/huggingface/nanotron/issues/78#issue-2147747937,

I tried to convert the weights from HF to nanotron format, but the generated outputs do not match those obtained using HF API. I ensured that converting these nanotron weights back to hf gives the same output as the original weights. The conversion was trivial except for the qkv and gate_up matrices, which are clubbed together as one parameter in nanotron. I assumed they are concatenated in the order (q,k,v) along the first dimension. 

The errors could be the following (assuming that the generation code is correct):
1. The order of qkv or up_gate parameters is different.
2. The rotary embeddings are different.

I saw the rotary embeddings of HF and Nanotron, and there is a slight difference between the two up to the 2nd decimal place. Could this be the reason for the inconsistency?

I would appreciate your help.

**Comment by xrsrke** [comment]
@NouamaneTazi do we have a conversion script from `transformers` to nanotron checkpoint?

**Comment by wiseyy** [comment]
Any updates? @xrsrke

**Comment by yardenas** [comment]
@wiseyy I'm facing a similar challenge. Any way we can join forces on this and try to make it work? :)

**Comment by yardenas** [comment]
@wiseyy unfortunately I can't go the megatron route (I'm part of a group and we already committed ourselves to nanotron).

> Conversion is straightforward

Can you help me get started with this? Maybe if I can reproduce your errors I'll be able to dig deeper into the issue

**Comment by xrsrke** [comment]
hello. you could use this https://github.com/huggingface/nanotron/tree/main/examples/llama

**Comment by wilzh40** [comment]
I've noticed the test about consistent logits are commented out for the above conversion scripts: https://github.com/huggingface/nanotron/blob/03d67f2103d5be0dc15ea6022a6cf16d6a633064/examples/llama/tests/test_conversion.py#L223

Also running into this problem of differing logits - any potential solutions? 

Also, given a nanotron checkpoint, how do we continue training on it? The above examples only show loading a model for inference but not for continued training.
